paasta_tools/cli/cmds/spark_run.py

```diff
@@ -423,8 +424,15 @@ def get_spark_config(
         user_args["spark.eventLog.enabled"] = "true"
         user_args["spark.eventLog.dir"] = default_event_log_dir
 
+    try:
+        mesos_address = find_mesos_leader(args.cluster)
+    except MesosLeaderUnavailable:
+        print(
+            "Couldn't reach the Mesos leader from here. Are you running this command from the right environment?",
```

Comment:
nit: "right environment" may not be obvious enough for developers. I'd put something like "(e.g. attempting to access prod from dev)".